### ISSUE TITLE
When the 2 knobs are on the lowest value

### ISSUE THREAD

**Issue by tappella** [chicio/RangeUISlider#3]
Hi,
first of all, congratulations for the amazing job!

I would like to share one thing that I came across while using your RangeUISlider. I'm using it as a Age Ranger control and the values could be the same, no problem until then. However, if both knobs are set to the minimum value there is a problem because the first responder on this case is the left knob and there is no room for it to move.

I ended up by doing a dirty fix on the code for now as below.

on the moveLeftKnob function, I included the else on your if clause:

```
if positionForKnob >= 0 && positionForKnob <= positionRightKnob {
    self.leftKnob.xPositionConstraint.constant = positionForKnob
}else{
    if gestureRecognizer.state == .began && positionForKnob <= 30 && positionRightKnob <= 10 {
        self.rightKnob.xPositionConstraint.constant = self.rightKnob.xPositionConstraint.constant + 100
    }
}
```

So, I basically check if it is starting the touch, check the finger position and if the right knob is so close to the left. With that, I push the right knob further to give the left one some space.

Thank you once again for the amazing piece of code.

Thiago

**Comment by chicio** [comment]
Hi @tappella,
thank you for your support!! I will try to investigate the problem during next  weekends.
Hope to find a solution quickly.
In the meantime, support my work by adding this repo to you favourite by starring it (and also other repo on my profile if you're interested).

Thank you again,
Fabrizio.

**Comment by chicio** [comment]
Hi @tappella,
I finally fixed the problem you mentioned in this issue. The Issue has been fixed in [1a31ae231ace9e88f47420ec5d1c6ea1316a50d7]. I pushed a new version on cocoapods because I also introduced other improvement:

- better code style
- better code structure
- upgrade to XCode 9 and Swift 4

Hope you will find it useful.
Thank you for your support!
Fabrizio.
